web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js

```diff
@@ -288,11 +288,24 @@ odoo.define('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer', function (requ
             }
             // Enforce mode of the parent
             options.mode = this.getParent().mode;
-            var widget = this._renderFieldWidget(
-                node, record, _.pick(options, 'mode')
-            );
-            this._handleAttributes(widget.$el, node);
-            return $td.append(widget.$el);
+            if (node.tag === 'widget') {
```

Comment:
TBH I don't remember why I discarded this stuff during refactoring :thinking: 
Fact is: it looks good and it's the way the std list renderer uses it :ok:

Reply:
Who knows :) Also, button seems to have no meaning fir this renderer, right?

Reply:
not sure we need buttons here but leaving it won't hurt as long as it does not fail.

Reply:
It won't render anyway, thus no need of keeping this unless adding support for buttons